docs/source/daml-integration-kit/index.rst

```diff
@@ -0,0 +1,371 @@
+  they need to integrate DAML with their ledger of choice completely on their
+  own.
+
+  Before reaching BETA status, we expect to have
+
+  - hardened our test tooling,
+  - built tooling for benchmarking DAML ledgers,
+  - completed several integrations of DAML for different ledgers, and
+  - implemented uniform *administrative DAML ledger access* to provide a
+    portable way for testing DAML applications against different DAML ledgers.
+
+GA (`tracking GitHub issue <https://github.com/digital-asset/daml/issues/661>`__)
+  For us GA (General Availability) means that there are several
+  production-ready DAML ledgers built using the DAML Integration Kit. We
+  expect to reach GA in 2019.
+
+
+DAML Ledger overview
+====================
+
+**TODO (ALPHA):** explain what ledgers exist; and what state they are in.
+(`GitHub issue <https://github.com/digital-asset/daml/issues/673>`__)
+
+
+.. _integration-kit_implementing:
+
+Implementing your own DAML Ledger
+*********************************
+
+
+Architectural overview
+======================
+
+We explain how to implement a DAML ledger backed by a specific
+ledger `X`. This can be a proper distributed ledger or also just a database.
+The goal of this implementation is to allow multiple DAML applications, which are potentially run by
+different entities, to execute multi-party workflows using the ledger `X`. We
+illustrate this in the following diagram.
+
+.. image:: images/architecture-overview.svg
+
+.. original: https://www.lucidchart.com/invitations/accept/69799877-4e80-444d-96a3-3e90814e94df
+
+We assume that the `X` ledger allows entities to participate in the
+evolution of the ledger via particular nodes. In the remainder of this
+documentation, we call these nodes `participant nodes`.
+
+The boxes labeled `daml-on-<X>-server` denote the DAML Ledger API
+servers, which implement the DAML Ledger API on top of the services provided
+by the `X` participant nodes.
+
+The boxes labeled `<X>-daml-validator` denote `X`-specific DAML transaction
+validation services. In a distributed ledger they provide the ability for
+nodes to :doc:`validate DAML transactions </concepts/ledger-model/ledger-integrity>`
+at the appropriate stage in the `X` ledger's transaction commit process.
+Whether they are needed, by what nodes they are used, and whether they are run
+in-process or out-of-process depends on the `X` ledger's architecture. Above
+we depict a common case where the participant nodes jointly maintain the
+ledger's integrity and therefore need to validate DAML transactions.
+
+
+Message flow
+------------
+
+**TODO (BETA):**
+
+- explain to readers the life of a transaction at a high-level, so they have a
+  mental framework in place when looking at the example code.
+  (`GitHub issue <https://github.com/digital-asset/daml/issues/672>`__)
+
+
+Implementing your server and validator
+======================================
+
+Each `X` ledger requires at least the implementation of a specific
```

Comment:
I think this content could do with its own heading - something like `Resources to help with ledger development`- and it can go further down, after the 'getting started' and 'deciding on the architecture' instructions. The reason being, those two sections are effectively acting as an index.